Riot/Modules/TabBar/MasterTabBarController.m

```diff
@@ -1034,6 +990,147 @@ - (void)promptUserBeforeUsingAnalytics
+    NSArray<MXDeviceInfo*> *devices = [session.crypto.store devicesForUser:session.myUserId].allValues;
+    
+    BOOL isUserHasOneUnverifiedDevice = NO;
+    
+    for (MXDeviceInfo *device in devices)
+    {
+        if (device.trustLevel.localVerificationStatus == MXDeviceUnknown)
+        {
+            isUserHasOneUnverifiedDevice = YES;
+            break;
+        }
+    }
+    
+    if (isUserHasOneUnverifiedDevice)
+    {
+        self.reviewSessionAlertHasBeenDisplayed = YES;
+        [self presentReviewUnverifiedSessionsAlertWithSession:session];
+    }
+}
+
+- (void)presentReviewUnverifiedSessionsAlertWithSession:(MXSession*)session
+{
+    [currentAlert dismissViewControllerAnimated:NO completion:nil];
+    
+    UIAlertController *alert = [UIAlertController alertControllerWithTitle:NSLocalizedStringFromTable(@"key_verification_self_verify_unverified_sessions_alert_title", @"Vector", nil)
+                                                                   message:NSLocalizedStringFromTable(@"key_verification_self_verify_unverified_sessions_alert_message", @"Vector", nil)
+                                                            preferredStyle:UIAlertControllerStyleAlert];
+    
+    [alert addAction:[UIAlertAction actionWithTitle:NSLocalizedStringFromTable(@"key_verification_self_verify_unverified_sessions_alert_validate_action", @"Vector", nil)
+                                              style:UIAlertActionStyleDefault
+                                            handler:^(UIAlertAction * action) {
+                                                [self showSettingsSecurityScreenForSession:session];
+                                            }]];
+    
+    [alert addAction:[UIAlertAction actionWithTitle:NSLocalizedStringFromTable(@"later", @"Vector", nil)
+                                              style:UIAlertActionStyleCancel
+                                            handler:nil]];
+    
+    [alert addAction:[UIAlertAction actionWithTitle:NSLocalizedStringFromTable(@"do_not_ask_again", @"Vector", nil)
+                                              style:UIAlertActionStyleDestructive
+                                            handler:^(UIAlertAction * action) {
+                                                RiotSettings.shared.hideReviewSessionsAlert = YES;
+                                            }]];
+    
+    
+    [self presentViewController:alert animated:YES completion:nil];
+    
+    currentAlert = alert;
+}
+
+- (void)showSettingsSecurityScreenForSession:(MXSession*)session
+{
+    SettingsViewController *settingsViewController = [SettingsViewController instantiate];
+    [settingsViewController loadViewIfNeeded];
+    SecurityViewController *securityViewController = [SecurityViewController instantiateWithMatrixSession:session];
+    
+    [[AppDelegate theDelegate] restoreInitialDisplay:^{
+        self.navigationController.viewControllers = @[settingsViewController, securityViewController];
```

Comment:
You cannot leave the settings once it is displayed.
There is no back button on the SettingsVC. Should we add self in this array?